packages/utils/src/store/ReactStore.ts

```diff
@@ -134,6 +170,16 @@ export class ReactStore<
    * @param value The new value to set for the specified key.
    */
   public set<T>(key: keyof State, value: T): void {
+    const interceptor = this.writeInterceptors.get(key);
+    if (interceptor) {
+      const updatedValue = interceptor(value, this);
+      if (!this.controlledValues.get(key) === true) {
+        super.set(key, updatedValue);
+      }
+
+      return;
+    }
```

Comment:
This implementation makes a hashmap check for every write even if no `writeInterceptors` were provided. It would be better for performance to keep `writeInterceptors` empty and avoid the `Map.get()` call.

This also applies for the other write methods.